routing/missioncontrol.go

```diff
@@ -421,6 +412,9 @@ func (m *MissionControl) processPaymentResult(result *paymentResult) (
 		return nil, err
 	}
 
+	m.Lock()
```

Comment:
Seems like this introduces the possibility of the following interleaving:
```
store.AddResult(result1)
store.AddResult(result2)
applyPaymentResult(result2)
applyPaymentResult(result1)
```

But maybe that's okay, since the timestamps ensure eventual consistency of the mission control state?

IMO a lock-less `getProbability` that can be called from inside the db transaction might be preferable, but this fix is simple and if the above is true i think the tradeoff is acceptable 👍

Reply:
ah i was thinking more about the in-memory state, but i hadn't considered the db :P

looking now, i was assuming we would ignore results that occur _before_ the current result, but it doesn't appear that way:
https://github.com/lightningnetwork/lnd/blob/master/routing/missioncontrol_state.go#L71

Assuming `result1` and `result2` are both failures or successes, Ideally the following:
 ```
 applyPaymentResult(result1)
 applyPaymentResult(result2)
```
and
```
applyPaymentResult(result2)
applyPaymentResult(result1)
```
would result in the same outcome (`result2`) for the failure/success.

I think there could also be some edge cases around handling second chances, since that's the only other that jumps out as being dependent on ordering. It might not be as big of an issue though since you have the payment lifecycle separating the write to disk.